### ISSUE TITLE
Pre-built binaries

### ISSUE THREAD

**Issue by JoCat** [Winterreisender/webview-nodejs#29]
I suggest adding a binary build for the module in order to eliminate unnecessary build steps on the users' side and to make it possible to conveniently work with the package as dependencies in other projects.

**Comment by songyz2019** [comment]
In the earlier version(maybe 0.1), there were prebuilt binaries instead of building during installation.  
I changed the plan to build during installation  because:
1. It's the common practice in node.js
2. It's hard to make .node file for every platform and every node.js version 
3. If we use prebuilt version, there's 3 choices: node-ffi-napi is not actively maintained, koffi has a serious bug that cannot be solved and node:ffi is experimental at that time.
4. the webview/webview provide a swig file so that we can easily make a node napi extension.

So, do we need try to provide prebuilt binaries now? If I have time, I'd like to try, especially node:ffi to see if it works, because webview_deno is using prebuilt binaries with deno's built-in ffi api, that's a very popular project for deno and we should follow it.

**Comment by Winterreisender** [comment]
In the earlier version(maybe 0.1), there were prebuilt binaries instead of building during installation.
I changed the plan to build during installation because:


It's the common practice in node.js
It's hard to make .node file for every platform and every node.js version
If we use prebuilt version, there's 3 choices: node-ffi-napi is not actively maintained, koffi has a serious bug that cannot be solved and node:ffi is experimental at that time.
the webview/webview provide a swig file so that we can easily make a node napi extension.
So, do we need try to provide prebuilt binaries now? If I have time, I'd like to try, especially node:ffi to see if it works, because webview_deno is using prebuilt binaries with deno's built-in ffi api, that's a very popular project for deno and we should follow it.

**Comment by JoCat** [comment]
I think it's worth a try 😉

**Comment by Winterreisender** [comment]
TLDR: I decided make pre-build `.node` binaries. I only found out an hour ago that this is possible.

It seems that the packager resolve the relative path of `.node` files in a wrong way. Since [node sea](https://nodejs.org/api/single-executable-applications.html) is an experimental feature, I think, maybe ...... it's not a good time to fix it before it becomes stable.  

And after thinks for a long time, I decide to keep using Node-API to build `.node` files and try to distribute pre-built `.node` files. Yes, I only knew [it is possible](https://github.com/nodejs/node-addon-api/blob/main/doc/prebuild_tools.md) one hour ago.   

And I finally decided not to follow webview-bun and webview-deno to make pre-built `.dll`,`.so`. The reason is, well, there's still no reliable ffi libary for Node.js I can find for webview.  I have to say that Node.js really falls behind Bun and Deno in FFI which makes webview-nodejs can only distribute pre-built `.node` files instead of more common used `.dll`,`.so`.

**Comment by JoCat** [comment]
As far as I remember, .node files are the same .dll and .so, just renamed. I've now come to the point where I'm writing a [wrapper](https://github.com/AuroraTeam/webview) for my project that uses wry and tao (tauri libraries) under the hood to create windows for nodejs. And I'm building it using napi-rs. You might find this option suitable for you.

**Comment by JoCat** [comment]
And about the node sea - yes, I didn't notice that right away. In this case you can use bpkg, it will package .node files in base64 and load them into runtime. But in my case it doesn't work because it requires modification, and it's probably easier to write a plugin for esbuild (I use it in my project), which will put the .node files next to the program executable.
